services/usersStatusService.ts

```diff
@@ -0,0 +1,26 @@
+const { generateAuthTokenForCloudflare } = require("../utils/discord-actions");
+
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+const updateDiscordUserNickname = async (discordId: string, userName: string) => {
+  try {
+    const authToken = generateAuthTokenForCloudflare();
+    const response = await fetch(`${DISCORD_BASE_URL}/guild/member`, {
+      method: "PATCH",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+      body: JSON.stringify({ discordId, userName }),
+    });
+    if (!response.ok) {
+      throw new Error("Could not update nickname");
+    }
+
+    await response.text();
```

Comment:
do we need this if we are not returning. we can remove it if we don't require it.